implementations/rust/ockam/ockam_command/src/node/show.rs

```diff
@@ -31,20 +33,89 @@ pub struct ShowCommand {
 
 impl ShowCommand {
     pub fn run(self, options: CommandGlobalOpts) {
-        let cfg = &options.config;
-        let port = match cfg.inner().nodes.get(&self.node_name) {
-            Some(cfg) => cfg.port(),
-            None => {
-                eprintln!("No such node available.  Run `ockam node list` to list available nodes");
-                std::process::exit(exitcode::IOERR);
+        node_rpc(run_impl, (options, self))
+    }
+}
+
+async fn run_impl(
+    ctx: ockam::Context,
+    (opts, cmd): (CommandGlobalOpts, ShowCommand),
+) -> crate::Result<()> {
+    let node_name = cmd.node_name;
+    let cfg = &opts.config;
+    let node_cfg = cfg.get_node(&node_name)?;
+    let tcp = TcpTransport::create(&ctx).await?;
+    let port = match cfg.inner().nodes.get(&node_name) {
+        None => {
+            eprintln!("No such node available.  Run `ockam node list` to list available nodes");
+            return Err(crate::error::Error::new(
+                exitcode::IOERR,
+                anyhow!("No nodes available for node show command"),
+            ));
+        }
+        Some(x) => x.port(),
+    };
+    let mut base_route = route![(TCP, format!("localhost:{}", port))];
+    let route = base_route.modify().append(NODEMANAGER_ADDR).into();
+    let mut child_ctx = ctx.new_detached(Address::random_local()).await?;
```

Comment:
You don't really need this. You can make the `run_impl` context mutable and use it on the `is_node_up` function.